Python/Product/Analysis/ProjectEntry.cs

```diff
@@ -56,8 +56,11 @@ IAnalysisCookie cookie
             DocumentUri = documentUri ?? MakeDocumentUri(filePath);
             FilePath = filePath;
             Cookie = cookie;
+
             MyScope = new ModuleInfo(ModuleName, this, state.Interpreter.CreateModuleContext());
             _unit = new AnalysisUnit(null, MyScope.Scope);
+            Analysis = new ModuleAnalysis(_unit, ((ModuleScope)_unit.Scope));
```

Comment:
Is this why you're ending up with the unit with null Ast? Aren't we checking that `Analysis` isn't null everywhere already?

Reply:
Right, but that `_unit` would normally never have ended up in `Analysis`, which means `FindScope` would not have been called on it. The change here is what causes that, so I'm wondering why we want this change if it breaks things and we're checking for null `Analysis` already?

Reply:
This was for #4138 which we took out.